fastq/tenXGenomics.go

```diff
@@ -37,9 +37,10 @@ func NextSeq(fwdReader *fileio.EasyReader, revReader *fileio.EasyReader) (*Linke
 //2) Next 6 bases is a 6 base Umi
 //3) finally adaptors and genomic sequence
 //TODO: consider trimming off adapter sequences too
+//TODO: it looks like tenX is now using a 10bp UMI, maybe an option to handle that is needed?
```

Comment:
I agree with changing 17 to 16, but not 22 to 24. But to my understand the linked construct is (16+6+128=150)

Reply:
I don't think there is a 22 -> 24 change being proposed, but maybe I am not seeing it right now.  Did you mean 24 -> 22?  The 24 -> 22 change (old line 42, new line 43) is because I am thinking that the barcode + umi is 22 bases (16+6=22), so the 23rd base is index 22 (zero based).
There are three changes I am proposing:
1) 17 -> 16 for the start of the UMI (inclusive)
2) 23 -> 22 for the end of the UMI (not inclusive)
3) 24 -> 22 for the start of the non-barcode, non-umi read (inclusive)
I am not an expert in the TenX tech, but I think we are on the same page with the first 16 being barcode and the next 6 being the UMI.  I think we just need to make sure we align on the same starts and stops to have the computer parse out the bases we want.

Reply:
Yes, I think I wrote that flipped. I think in my head I thought that was changing the 6bp UMI to 10 bp, but now that I look at the math again I think the 16+6 = 22 is right.

Ok I am on board for the changes, but maybe the comment on line 40 should change to say something like this is for 10x linked read and not single cell 10x libraries.

We can certainly do something like pass in 6 or 10 bp for the UMI, but I think in order to avoid confusion the LinkedRead struct should be then renamed to something like TenXReads{} (or something a long that line)